### PR TITLE
Push notifications reports

### PR DESCRIPTION
**Pull request by hcarneiro**

### CODE REVIEW

File: js/interface.js [hunk 1]
```diff
@@ -1,13 +1,104 @@
+var $tbody = $('#jobs-entries');
+var source = $('#template-table-entries').html();
+var template = Handlebars.compile(source);
+
+var tempJob = {
```
Comment (squallstar) [inline js/interface.js]: no variables should be named `temp`. ever :)

File: js/interface.js [hunk 2]
```diff
@@ -1,13 +1,104 @@
+var $tbody = $('#jobs-entries');
+var source = $('#template-table-entries').html();
+var template = Handlebars.compile(source);
```
Comment (squallstar) [inline js/interface.js]: template is too generic as a word (since we might be adding more)

File: js/interface.js [hunk 3]
```diff
@@ -1,13 +1,104 @@
...
+      tempJob.sentGoogle = gcm;
+      tempJob.sentApple = apn;
+      tempJob.sentWindows = wns;
+      tempJob.dataSourceName = job.dataSourceEntry.dataSource ? job.dataSourceEntry.dataSource.name : "Data source was deleted";
+
+      reportData.jobs.push(tempJob);
+    });
+
+    if (reportData.jobs.length) {
+      var html = template(reportData);
```
Comment (squallstar) [inline js/interface.js]: no `var` should be declared inside conditional blocks

Comment (hcarneiro) [inline js/interface.js]: Why declare it if there's a possibility of not using it?

Comment (squallstar) [inline js/interface.js]: what I'm saying is that you should never declare variables inside `if` blocks. This should be done instead:

```js
var html;

if (reportData.jobs.length) {
  html = template(reportData);
  ...
}
```

Comment (squallstar) [inline js/interface.js]: btw I think you already did it on the last code change :)

File: js/interface.js [hunk 4]
```diff
@@ -63,26 +155,35 @@ var UINotification = (function() {
...
+    messageCharLimit: 235,
+    notificationConfig: {},
+    sendErrorMessage: '',
+    mockedRequest: Fliplet.Env.get('development') // Use a mocked request under development environment
   };
 
   UINotification.prototype.initUI = function() {
+    Fliplet.App.Subscriptions.get().then(function(subscriptions) {
+      if (subscriptions.length === 0) {
+        $('#subscription-note').html('There are no registered devices to receive this notification.');
```
Comment (tonytlwu) [inline js/interface.js]: `There are no registered devices to receive this notification.` => `There are no devices registered to receive this notification.`

File: interface.html [hunk 5]
```diff
@@ -16,98 +17,160 @@ <h1 class="text-center">Push Notifications</h1>
...
+          </div>
+          <div class="col-sm-12">
+            <textarea class="form-control" name="popup_message" id="popup_message" rows="4" placeholder="Enter a message">{{#if popupMessage}}{{popupMessage}}{{else}}You will receive notifications when app updates are available.
+Please tap on Allow Notifications below to receive notifications on the updates.{{/if}}</textarea>
+          </div>
+        </div>
+        <div class="form-group clearfix checkbox">
+          <div class="col-sm-12">
+            <div class="checkbox checkbox-icon">
+              <input type="checkbox" id="auto-popup" name="showAutomatically" value="1" {{#if showAutomatically}}checked{{/if}} />
```
Comment (ibroom) [inline interface.html]: Can we add a message that says 'By default the first screen of the app will ask the user to accept push notifications. If you want to manually add the code to a specific screen, please disable this option.'

File: interface.html [hunk 6]
```diff
@@ -16,98 +17,160 @@ <h1 class="text-center">Push Notifications</h1>
...
-      					<small class="text-info"><span class="glyphicon glyphicon-chevron-down"></span> Scroll down for Step 2</small>
-      				</div>
-      			</div>
+            </div>
           </div>
           <div class="row">
             <div class="col-xs-8 col-xs-offset-2 text-center">
+              <p id="subscription-note">Up to <strong><span id="subscriptions">x</span></strong> registered devices will recieve this notification.</p>
               <p>
                 <a class="btn btn-primary btn-lg notification-send" href="#">Send notification</a>
```
Comment (ibroom) [inline interface.html]: @tonytlwu as discussed I suggest we add above the 'Send notification' button the following text:
'This app has X number of subscribed devices*. Your notification will be sent to all of them.'

Under the 'send notification' button in small text we should have:
'* Subscribed devices is the maximum number of devices ever subscribed and it is unlikly all devices will still be active push subscribers and see your message. This is a limitation of the push notification system.'

Comment (ibroom) [inline interface.html]: resolved, I didn't see it in the code.

**Pull request merged**